### PR TITLE
Re-work mask export flow to support state saving

### PR DESCRIPTION
**Pull request by bnmajor**

### CODE REVIEW

File: hexrd/ui/mask_manager_dialog.py [hunk 1]
```diff
@@ -202,34 +202,36 @@ def context_menu_event(self, event):
...
+            'HDF5 files (*.h5 *.hdf5)')
+
+        if not output_file:
+            return
+
+        HexrdConfig().working_dir = os.path.dirname(output_file)
+        # write to hdf5
+        with h5py.File(output_file, 'w') as f:
+            h5py_group = f.create_group('masks')
+            unwrap_dict_to_h5(h5py_group, data, asattr=False)
```
Comment (psavery) [inline hexrd/ui/mask_manager_dialog.py:231]:
```suggestion
            self.write_masks_to_group(data, h5py_group)
```

File: hexrd/ui/mask_manager_dialog.py [hunk 2]
```diff
@@ -202,34 +202,36 @@ def context_menu_event(self, event):
...
-            HexrdConfig().working_dir = os.path.dirname(output_file)
-            # write to hdf5
-            with h5py.File(output_file, 'w') as f:
-                h5py_group = f.create_group('masks')
-                unwrap_dict_to_h5(h5py_group, data, asattr=False)
-        else:
-            unwrap_dict_to_h5(h5py_group, data, asattr=False)
-
-    def export_visible_masks(self):
+    def gather_visible_masks(self, h5py_group=None):
```
Comment (psavery) [inline hexrd/ui/mask_manager_dialog.py:207]:
```suggestion
    def write_visible_masks(self, h5py_group=None):
```

File: hexrd/ui/mask_manager_dialog.py (hunk 1 above)
Comment (bnmajor) [reply hexrd/ui/mask_manager_dialog.py]: Changed

File: hexrd/ui/mask_manager_dialog.py (hunk 2 above)
Comment (bnmajor) [reply hexrd/ui/mask_manager_dialog.py]: Changed

File: hexrd/ui/mask_manager_dialog.py [hunk 3]
```diff
@@ -72,8 +72,8 @@ def setup_connections(self):
         self.ui.masks_table.cellDoubleClicked.connect(self.get_old_name)
         self.ui.masks_table.cellChanged.connect(self.update_mask_name)
         self.ui.masks_table.customContextMenuRequested.connect(
-            self.export_masks)
-        self.ui.export_masks.clicked.connect(self.export_masks)
+            self.context_menu_event)
+        self.ui.export_masks.clicked.connect(self.gather_visible_masks)
```
Comment (psavery) [inline hexrd/ui/mask_manager_dialog.py:76]: Need to change it here as well

File: hexrd/ui/mask_manager_dialog.py [hunk 4]
```diff
@@ -198,38 +198,41 @@ def context_menu_event(self, event):
             if action == export:
                 selection = self.ui.masks_table.item(index.row(), 0).text()
                 data = HexrdConfig().raw_mask_coords[selection]
-                d = {}
+                d = {'visible': []}
```
Comment (psavery) [inline hexrd/ui/mask_manager_dialog.py:201]:
```suggestion
                d = {'_visible': []}
```

File: hexrd/ui/mask_manager_dialog.py [hunk 5]
```diff
@@ -198,38 +198,41 @@ def context_menu_event(self, event):
...
-                unwrap_dict_to_h5(h5py_group, data, asattr=False)
-        else:
-            unwrap_dict_to_h5(h5py_group, data, asattr=False)
-
-    def export_visible_masks(self):
-        d = {}
-        for name in HexrdConfig().visible_masks:
-            data = HexrdConfig().raw_mask_coords[name]
+    def write_all_masks(self, h5py_group=None):
+        d = {'visible': HexrdConfig().visible_masks}
```
Comment (psavery) [inline hexrd/ui/mask_manager_dialog.py:210]:
```suggestion
        d = {'_visible': HexrdConfig().visible_masks}
```

File: hexrd/ui/mask_manager_dialog.py (hunk 4 above)
Comment (bnmajor) [reply hexrd/ui/mask_manager_dialog.py]: Fixed

File: hexrd/ui/mask_manager_dialog.py (hunk 3 above)
Comment (bnmajor) [reply hexrd/ui/mask_manager_dialog.py]: Fixed

File: hexrd/ui/mask_manager_dialog.py (hunk 5 above)
Comment (bnmajor) [reply hexrd/ui/mask_manager_dialog.py]: Done

**Review by psavery — APPROVED** [review]
LGTM

**Pull request merged**